src/extensions/csfs.rs

```diff
@@ -0,0 +1,266 @@
+//! Miniscript extension: ver_eq
+//! Note that this fragment is only supported for Segwit context
+//! You are most likely looking for taproot direct tx introspection
+
+use std::fmt;
+use std::str::FromStr;
+
+use elements::{self, opcodes, secp256k1_zkp};
+
+use crate::descriptor::CovError;
+use crate::miniscript::context::ScriptContextError;
+use crate::miniscript::decode::ParseableKey;
+use crate::miniscript::lex::{Token as Tk, TokenIter};
+use crate::miniscript::satisfy::{Satisfaction, Witness};
+use crate::miniscript::types::extra_props::{OpLimits, TimelockInfo};
+use crate::miniscript::types::{Base, Correctness, Dissat, ExtData, Input, Malleability};
+use crate::policy::{self, Liftable};
+use crate::util::MsKeyBuilder;
+use crate::{
+    expression, interpreter, miniscript, Error, Extension, ForEach, MiniscriptKey, Satisfier, Tap,
+    ToPublicKey, TranslatePk, Translator,
+};
+
+/// Version struct
+/// `DEPTH <12> SUB PICK <num> EQUAL`
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Clone, Copy)]
+pub struct CheckSigFromStack<Pk: MiniscriptKey> {
+    pub pk: Pk,
```

Comment:
In 761799269e0b5bf29a7a9fa6162a468653a5f351:

Should this fragment also enforce the signed message? In which case we'd need another field in this struct and updates to all the script serialization etc..